indexd_test_utils2/__init__.py

```diff
@@ -0,0 +1,296 @@
+    """
+    pass
+
+
+@pytest.fixture
+def indexd_client(indexd_server, create_indexd_tables, indexd_admin_user):
+    """Create the tables and add an auth user"""
+    return IndexClient(indexd_server.baseurl, auth=(indexd_admin_user[0], indexd_admin_user[1]))
+
+
+@pytest.fixture(scope='session')
+def indexd_server(pg_url):
+    """
+    Starts the indexd server, and cleans up its mess.
+    Most tests will use the client which stems from this
+    server fixture.
+
+    Runs once per test session.
+    """
+    app = flask.Flask("indexd")
+    settings = indexd_settings.get_settings(pg_url)
+    app_init(app, settings)
+
+    hostname = 'localhost'
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    debug = False
+
+    port = random.randint(8000, 9000)
+    while sock.connect_ex((hostname, port)) == 0:
+        port = random.randint(8000, 9000)
+
+    t = threading.Thread(target=app.run, kwargs={'host': hostname, 'port': port, 'debug': debug})
+    t.setDaemon(True)
+    t.start()
+
+    wait_for_indexd_alive(port)
+    yield MockServer(port=port)
+
+
+def wait_for_indexd_alive(port):
+    url = f'http://localhost:{port}'
+    try:
+        requests.get(url)
+    except requests.ConnectionError:
+        return wait_for_indexd_alive(port)
+    else:
+        return
+
+
+class MockServer:
+    def __init__(self, port):
+        self.port = port
+        self.baseurl = f'http://localhost:{port}'
+
+
+def create_random_index(index_client, did=None, version=None, hashes=None):
+    """
+    Shorthand for creating new index entries for test purposes.
+    Note:
+        Expects index client v1.5.2 and above
+    Args:
+        index_client (indexclient.client.IndexClient): pytest fixture for index_client
+        passed from actual test functions
+        did (str): if specified it will be used as document did, else allows indexd to create one
+        version (str): version of the index being added
+        hashes (dict): hashes to store on the index, if not specified a random one is created
+    Returns:
+        indexclient.client.Document: the document just created
+    """
+
+    did = str(uuid.uuid4()) if did is None else did
+
+    if not hashes:
+        md5_hasher = hashlib.md5()
+        md5_hasher.update(did.encode("utf-8"))
+        hashes = {'md5': md5_hasher.hexdigest()}
+
+    doc = index_client.create(
+        did=did,
+        hashes=hashes,
+        size=random.randint(10, 1000),
+        version=version,
+        acl=["a", "b"],
+        file_name=f"{did}_warning_huge_file.svs",
+        urls=[f"s3://super-safe.com/{did}_warning_huge_file.svs"],
+        urls_metadata={f"s3://super-safe.com/{did}_warning_huge_file.svs": {"a": "b"}}
+    )
+
+    return doc
+
+
+def create_random_index_version(index_client, did, version_did=None, version=None):
+    """
+    Shorthand for creating a dummy version of an existing index, use wisely as it does not assume any versioning
```

Comment:
This is a great call out in the docstring on intended use and potential concerns!

Reply:
This just creates fake indexd versions. Not sure there is any concern to use this for test.